arches/app/media/js/viewmodels/workflow.js

```diff
@@ -440,6 +434,16 @@ define([
             });
 
             window.location.assign(self.quitUrl);
+        }
+
+        this.finishWorkflow = function() {
+            if (self.activeStep().hasUnsavedData()) {
+                self.activeStep().save().then(function() {
+                    self.markWorkflowComplete();
+                });
```

Comment:
nit: this can be reduced to `self.activeStep().save().then(self.markWorkflowComplete)`